### PR TITLE
Add sokt support

### PR DESCRIPTION
**Pull request by josh-richardson**
Adds sokt support to the gradle plugin. @AlexandrouR should take a look at this and perhaps add some more commits to fix any issues that @xaviarias might have with this - first and foremost, I'm currently only checking the first source file to determine the required solidity version.

### CODE REVIEW

**Comment by xaviarias** [comment]
@josh-richardson is this the same strategy as in the Gradle plugin? 

Everything looks good to me, I just think would be good to have a similar approach when determining the version.

File: src/main/java/org/web3j/mavenplugin/JavaClassGeneratorMojo.java [hunk 1]
```diff
@@ -228,9 +227,6 @@ private String parseSoliditySources(Collection<String> includedFiles) throws Moj
                 SolidityCompiler.Options.METADATA
         );
         if (result.isFailed()) {
-            if (result.errors.contains("Source file requires different compiler version")) {
-                throw new VersionMismatchException(SolidityCompiler.getInstance(getLog()).getUsedSolCVersion(), result.errors);
-            }
             throw new MojoExecutionException("Could not compile solidity files\n" + result.errors);
```
Comment (xaviarias) [inline src/main/java/org/web3j/mavenplugin/JavaClassGeneratorMojo.java:230]: Maybe replace `solidity` by `Solidity` for clarity.

Comment (AlexandrouR) [reply src/main/java/org/web3j/mavenplugin/JavaClassGeneratorMojo.java]: Fixed

File: pom.xml [hunk 2]
```diff
@@ -3,7 +3,7 @@
     <groupId>org.web3j</groupId>
     <artifactId>web3j-maven-plugin</artifactId>
     <packaging>maven-plugin</packaging>
-    <version>4.5.12-SNAPSHOT</version>
+    <version>4.6.12</version>
```
[the remark this answers is not in the document]
Comment (AlexandrouR) [reply pom.xml]: Fixed

**Review by xaviarias — APPROVED** [review]

**Comment by adpe** [comment]
Mhm, this dependency has now some negative results on this plugin and the Solidity compiler (solc) handling. As a result, there are several issues: 
- https://github.com/web3j/web3j-maven-plugin/issues/68
- https://github.com/web3j/web3j-maven-plugin/issues/87
- https://github.com/web3j/web3j-maven-plugin/issues/91

So, this MR https://github.com/web3j/web3j-sokt/pull/10 should (issue https://github.com/web3j/web3j-sokt/issues/6) solve the issue. But `web3j-sokt` hasn't got a new release with these changes. And afterwards, you guys must update it here as well.

Why do you switch from `org.ethereum/solcJ-all` to this dependency for solc? What are the benefits?

Cheers,
Adrian

**Pull request merged**